aiohttp/web_utils.py

```diff
@@ -0,0 +1,57 @@
+import asyncio
+
+from aiohttp.web_exceptions import HTTPMovedPermanently, HTTPNotFound
+from aiohttp.web_urldispatcher import SystemRoute
+
+__all__ = ('normalize_path_middleware',)
+
+
+def normalize_path_middleware(*, merge_slashes=True, append_slash=True,
+                              redirect_class=HTTPMovedPermanently):
+
+    @asyncio.coroutine
+    def factory(app, handler):
+
+        @asyncio.coroutine
+        def middleware(request):
+            try:
+                return (yield from handler(request))
+            except HTTPNotFound as exc:
```

Comment:
Checking that `request.match_info` is instance of `_NotFoundMatchInfo` might resolve this issue.